packages/jsii-pacmak/lib/targets/go/runtime/property-access.ts

```diff
@@ -14,12 +15,24 @@ export class GetProperty {
 
   public emit(code: CodeMaker) {
     const resultVar = slugify('returns', [this.parent.instanceArg]);
+    const implMapVar = slugify('implMap', [this.parent.instanceArg]);
     code.line(`var ${resultVar} ${this.parent.returnType}`);
+    code.line(`${implMapVar} := make(${JSII_IMPL_MAP_TYPE})`);
+
+    const implMap =
+      this.parent.reference?.scopedImplMap(this.parent.parent.pkg) ?? [];
+    if (implMap.length) {
+      const [interfaceName, structName] = implMap;
+      code.line(
+        `${implMapVar}[reflect.TypeOf((*${interfaceName})(nil)).Elem()] = reflect.TypeOf((*${structName})(nil)).Elem()`,
+      );
```

Comment:
(Also below on line 68)

It would be nice to have a comment on this bad boy showing what a line it produces could look like... The amount of braces there is nauseating and it's kinda hard to picture what this produces.